numpy/core/_add_newdocs.py

```diff
@@ -7970,62 +7970,169 @@ def luf(lamdaexpr, *args, **kwargs):
+    except AttributeError:
+        return None
+
+def type_aliases(place, aliases):
+    def type_aliases_gen():
+        for (alias, doc) in aliases:
+            alias_type = get_type(place, alias)
+            if alias_type is not None:
+                yield (alias_type, alias, doc)
+    return list(type_aliases_gen())
+
+integer_aliases = type_aliases('numpy.core.numerictypes', [
+    ('int8', '8-bit signed integer (-128 to 127)'),
+    ('int16', '16-bit signed integer (-32768 to 32767)'),
+    ('int32', '32-bit signed integer (-2147483648 to 2147483647)'),
+    ('int64', '64-bit signed integer (-9223372036854775808 to 9223372036854775807)'),
+    ('intp', 'Signed integer large enough to fit pointer, compatible with C ``intptr_t``'),
+    ])
+
+unsigned_integer_aliases = type_aliases('numpy.core.numerictypes', [
+    ('uint8', '8-bit unsigned integer (0 to 255)'),
+    ('uint16', '16-bit unsigned integer (0 to 65535)'),
+    ('uint32', '32-bit unsigned integer (0 to 4294967295)'),
+    ('uint64', '64-bit unsigned integer (0 to 18446744073709551615)'),
+    ('uintp', 'Unsigned integer large enough to fit pointer, compatible with C ``uintptr_t``'),
+    ])
+
+float_aliases = type_aliases('numpy.core.numerictypes', [
+    ('float16', '16-bit-precision floating-point number type: sign bit, 5 bits exponent, 10 bits mantissa'),
+    ('float32', '32-bit-precision floating-point number type: sign bit, 8 bits exponent, 23 bits mantissa'),
+    ('float64', '64-bit precision floating-point number type: sign bit, 11 bits exponent, 52 bits mantissa'),
+    ('float96', '96-bit extended-precision floating-point number type'),
+    ('float128', '128-bit extended-precision floating-point number type'),
+    ])
+
+complex_aliases = type_aliases('numpy.core.numerictypes', [
+    ('complex64', 'Complex number type composed of 2 32-bit-precision floating-point numbers'),
+    ('complex128', 'Complex number type composed of 2 64-bit-precision floating-point numbers'),
+    ('complex192', 'Complex number type composed of 2 96-bit extended-precision floating-point numbers'),
+    ('complex256', 'Complex number type composed of 2 128-bit extended-precision floating-point numbers'),
+    ])
+
+
+def add_newdoc_for_numeric_type(obj, character_code, fixed_aliases, possible_aliases, doc):
+    place = 'numpy.core.numerictypes'
+    o = get_type(place, obj)
+    if o is None:
+        return
+
+    canonical_name_doc = "" if obj == o.__name__ else "Canonical name: ``{}``.\n".format(obj)
+    alias_doc = ''.join("Alias: ``{}``.\n    ".format(alias) for alias in fixed_aliases)
+    alias_doc += ''.join("Alias *on this platform*: ``{}``: {}.\n    ".format(alias, doc)
+                         for (alias_type, alias, doc) in possible_aliases if alias_type is o)
+
+    docstring = """
+    {doc}
+    Character code: ``'{character_code}'``.
+    {canonical_name_doc}{alias_doc}
+    """.format(doc=doc.strip(), character_code=character_code,
```

Comment:
You'll find the indentation a lot easier to manage if you use textwrap dedent here - in general you want to dedent before doing any formatting

Reply:
I'm a bit weary to dedent everything, because other docstrings seem to be indented as well?
E.g.: 

```pycon
>>> pprint.pprint(np.dtype.__doc__)
('dtype(obj, align=False, copy=False)\n'
 '\n'
 '    Create a data type object.\n'
 '\n'
 '    A numpy array is homogeneous, and contains elements described by a\n'
 '    dtype object. A dtype object can be constructed from different\n'
 '    combinations of fundamental numeric types.\n'
 '\n'
 '    Parameters\n'
 '    ----------\n'
 '    obj\n'
 '        Object to be converted to a data type object.\n'
[...]
```

Reply:
I wasn't aware of that, interesting find. I'd assumed that we dedented them in addnewdocs, but it seems that python does it automatically in `help`. I think ultimately the indentation doesn't matter, so what you have is fine.

Reply:
The whole pasting-strings-together is a bit of a mess; you're right there, but until there's more changes that need to be done, this might be the easiest